samtranslator/swagger/swagger.py

```diff
@@ -691,10 +696,15 @@ def add_auth_to_method(self, path: str, method_name: str, auth: Dict[str, Any],
         method_scopes = auth and auth.get("AuthorizationScopes")
         api_auth = api and api.get("Auth")
         authorizers = api_auth and api_auth.get("Authorizers")
+        required_options_api_key = method_name == "options" and (auth.get("AddApiKeyRequiredToCorsPreflight") is False)
```

Comment:
We should fail if it's not a boolean. Otherwise values such as `"false"` won't enable the setting, but will quietly transform without error.

You can use this:

https://github.com/aws/serverless-application-model/blob/1034e333cf1020a15a35329dd110446e9bc70cfb/samtranslator/validator/value_validator.py#L123-L128